Gems/Atom/RPI/Code/Source/RPI.Public/MeshDrawPacket.cpp

```diff
@@ -44,15 +44,19 @@ namespace AZ
         {
             if (!m_material)
             {
-                const ModelLod::Mesh& mesh = m_modelLod->GetMeshes()[m_modelLodMeshIndex];
-                m_material = mesh.m_material;
+                m_material = GetMesh().m_material;
             }
         }
 
         Data::Instance<Material> MeshDrawPacket::GetMaterial() const
         {
             return m_material;
         }
+        
+        const ModelLod::Mesh& MeshDrawPacket::GetMesh() const
+        {
+            return m_modelLod->GetMeshes()[m_modelLodMeshIndex];
```

Comment:
This should assert that the index is in range or return empty data if it's out of range.